crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -371,10 +402,87 @@ impl ReadOnlyUserIdentities {
 /// only contain a master key and a self signing key, meaning that only device
 /// signatures can be checked with this identity.
 #[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(from = "ReadOnlyUserIdentitySerializer", into = "ReadOnlyUserIdentitySerializer")]
 pub struct ReadOnlyUserIdentity {
     user_id: OwnedUserId,
     pub(crate) master_key: Arc<MasterPubkey>,
     self_signing_key: Arc<SelfSigningPubkey>,
+    /// The first time a cryptographic identity is seen for a given user, it
+    /// will be associated to that user (i.e pinned). Future interaction
+    /// will expect this user crypto identity to stay the same,
+    /// this will help prevent some MITM attacks.
+    /// In case of identity change, it will be possible to pin the new identity
+    /// is the user wants.
+    pinned_msk: Arc<RwLock<MasterPubkey>>,
```

Comment:
Worth noting the presence of this field means that the documentation on this class ("other users will
only contain a master key and a self signing key"), as well as its name, are no longer correct. We should at least fix the doc; fixing the name would be good too but might be a bit invasive (plus, we'd need to think of good names).